ui.frontend/src/GuideBridge.js

```diff
@@ -451,6 +451,84 @@ class GuideBridge {
+     * @param {string} [formContainerPath] - Optional path of the form container to unload. 
+     *        If not provided, unloads the currently connected form.
+     * @method
+     * @memberof GuideBridge
+     * @instance
+     * @example
+     * // Simplest - unload current connected form
+     * guideBridge.unloadAdaptiveForm();
+     * 
+     * // Unload specific form by path (for multiple forms on page)
+     * guideBridge.unloadAdaptiveForm('/content/forms/af/myform');
+     * 
+     * // Get path from form container and unload
+     * const formPath = formContainer.getPath();
+     * guideBridge.unloadAdaptiveForm(formPath);
+     * 
+     */
+    unloadAdaptiveForm(formContainerPath) {
+        // Use provided path or fall back to current formContainerPath
+        const pathToUnload = formContainerPath || this.#formContainerPath;
+        
+        if (!pathToUnload) {
+            console.warn("No form container path specified or available to unload.");
+            return;
+        }
+        
+        // Get the container element from the view
+        let container = null;
+        if (this.#formContainerViewMap[pathToUnload]) {
+            const formContainerView = this.#formContainerViewMap[pathToUnload];
+            container = formContainerView.getFormElement();
+        }
+
+        // Clean up widget elements appended to document.body
+        // These widgets are created by captcha components and datepicker that append to body
+        const widgetSelectors = [
+            '.cmp-adaptiveform-recaptcha__widget',
+            '.cmp-adaptiveform-turnstile__widget',
+            '.cmp-adaptiveform-hcaptcha__widget',
+            '.datePickerTarget'
+        ];
+        
+        widgetSelectors.forEach(selector => {
+            const widgets = document.querySelectorAll(selector);
+            widgets.forEach(widget => widget.remove());
+        });
+
+        // Remove DOM content if container is found
+        if (container) {
+            if (typeof container.replaceChildren === 'function') {
+                container.replaceChildren();
+            } else {
+                // Fallback for older browsers
+                container.innerHTML = '';
+            }
+        }
+
+        // Clear DatePicker cache
+        // afCache stores datepicker instances keyed by element IDs
+        if (window.afCache && typeof window.afCache.store === 'object') {
+            window.afCache.store = {};
+        }
+
+        // Remove the form container from the map
+        delete this.#formContainerViewMap[pathToUnload];
+
+        // Reset formContainerPath if it matches the one being unloaded
+        if (this.#formContainerPath === pathToUnload) {
+            this.#formContainerPath = "";
+        }
+
+        // Note: FunctionRuntime.customFunctions is intentionally NOT cleared
+        // as it's a global registry shared across all forms, matching classic behavior
+    }
```

Comment:
We also have mutationObservers ? We should clear those as well if not doing them

Reply:
Observer will eventually be garbage collected, but it's not immediate and could cause temporary memory overhead. Let me fix this as well